src/Symfony/Bridge/Doctrine/Form/DoctrineOrmTypeGuesser.php

```diff
@@ -95,7 +95,7 @@ public function guessRequired($class, $property)
 
         // Check whether the field exists and is nullable or not
         if ($classMetadata->hasField($property)) {
-            if (!$classMetadata->isNullable($property)) {
+            if (!$classMetadata->isNullable($property) && $classMetadata->getTypeOfField($property) !== 'boolean') {
```

Comment:
actually also use the constant please: https://github.com/doctrine/dbal/blob/master/lib/Doctrine/DBAL/Types/Type.php#L40